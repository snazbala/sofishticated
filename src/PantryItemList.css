.pantry-item-list {
    margin: 20px;
    padding: 20px;
    background: #00bcbc;
}