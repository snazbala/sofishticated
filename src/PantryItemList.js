import React, { Component } from 'react';
import AddPantryItem from './AddPantryItem';
import PantryItem from './PantryItem';
import './PantryItemList.css';

class PantryItemList extends Component {
    constructor(props) {
        super(props);
        this.state = { pantryItems: ['Flour', 'Sugar'] };
        this.addPantryItem = this.addPantryItem.bind(this);
        this.removePantryItem = this.removePantryItem.bind(this);
    }

    addPantryItem(newPantryItem) {
        this.setState({ pantryItems: [...this.state.pantryItems, newPantryItem] });
    }

    removePantryItem(itemToRemove) {
        const filteredPantryItems = this.state.pantryItems.filter(name => {
            return name !== itemToRemove;
        });
        this.setState({ pantryItems: filteredPantryItems });
    }

    renderPantryItems() {
        return this.state.pantryItems.map(name => (
            <PantryItem
                key={name}
                name={name}
                removePantryItem={this.removePantryItem}
            />
        ));
    }

    render() {
        return (
            <div className="pantry-item-list">
                <AddPantryItem addPantryItem={this.addPantryItem} />
                {this.renderPantryItems()}
            </div>
        );
    }
}

export default PantryItemList;