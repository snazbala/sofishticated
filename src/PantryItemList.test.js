import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import PantryItemList from './PantryItemList';

describe(PantryItemList, () => {
    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <PantryItemList/>
        )
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});


