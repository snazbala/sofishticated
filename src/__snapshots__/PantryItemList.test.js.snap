// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class PantryItemList extends _react.Component {
    constructor(props) {
        super(props);
        this.state = {
            items: [
            {
                name: 'Flour',
                quantity: '12' },

            {
                name: 'Sugar',
                quantity: '2' }] };



        this.addPantryItem = this.addPantryItem.bind(this);
        this.removePantryItem = this.removePantryItem.bind(this);
    }

    addPantryItem(newPantryItem) {
        this.setState({ items: [...this.state.items, newPantryItem] });
    }

    removePantryItem(itemToRemove) {
        const filteredPantryItems = this.state.items.filter(item => {
            return item.name !== itemToRemove;
        });
        this.setState({ items: filteredPantryItems });
    }

    renderPantryItems() {
        return this.state.items.map((item, index) =>
        _react2.default.createElement(_PantryItem2.default, {
            key: item.name + index,
            name: item.name,
            quantity: item.quantity,
            removePantryItem: this.removePantryItem, __source: { fileName: _jsxFileName, lineNumber: 38 }, __self: this }));


    }

    render() {
        return (
            _react2.default.createElement('div', { className: 'pantry-item-list', __source: { fileName: _jsxFileName, lineNumber: 49 }, __self: this },
                _react2.default.createElement(_AddPantryItem2.default, { addPantryItem: this.addPantryItem, __source: { fileName: _jsxFileName, lineNumber: 50 }, __self: this }),
                this.renderPantryItems()));


    }} renders and matches our snapshot 1`] = `
<div
  className="pantry-item-list"
>
  <div
    className="add-pantry-item"
  >
    <label
      className="label-padding"
    >
      Name
    </label>
    <input
      className="input-field"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      className="label-padding"
    >
      Quantity Remaining
    </label>
    <input
      className="input-field"
      name="quantity"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      onClick={[Function]}
    >
      Add New Item
    </button>
  </div>
  <div
    className="pantry-item"
  >
    Flour
     
    12
       
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
  <div
    className="pantry-item"
  >
    Sugar
     
    2
       
    <button
      className="remove"
      onClick={[Function]}
    >
      Remove
    </button>
  </div>
</div>
`;
